packages/e2e-tests/specs/editor/blocks/navigation.test.js

```diff
@@ -134,12 +134,5 @@ describe( 'Navigation', () => {
 
 		// Expect a Navigation Block with two Navigation Links in the snapshot.
 		expect( await getEditedPostContent() ).toMatchSnapshot();
-
-		// TODO - this is needed currently because when adding a link using the suggestion list,
-		// a submit button is used. The form that the submit button is in is unmounted when submission
-		// occurs, resulting in a warning 'Form submission canceled because the form is not connected'
-		// in Chrome.
-		// Ideally, the suggestions wouldn't be implemented using submit buttons.
-		expect( console ).toHaveWarned();
```

Comment:
Related: #19056

Also: #19551 (not sure yet if and how it relates to this effort)

Reply:
yes, this solves the first one (clicking a suggestion was removing the form but the same click was also submitting the form).

The second one, I believe we should just make sure LinkControl  (and its search subcomponent) replaces URLInput entirely instead of the weird relationship they have right now where almost everything is overridden with props.